ideascube/widgets.py

```diff
@@ -1,6 +1,56 @@
 from django.forms import widgets
 
 from ideascube.configuration import get_config
+from django.utils.html import mark_safe, escape
+from django.utils.encoding import force_text
+from django.forms.utils import flatatt
+
+
+_script_template = '''<script type="text/javascript">tinymce.init({{
+selector: "#{widget_id}",
+inline: true,
+menubar: {menubar},
+toolbar: {toolbar},
+plugins: "{plugins}",
+hidden_input: false
+}});</script>'''
+
+
+class RichTextEntry(widgets.Widget):
```

Comment:
True.

I was thinking in terms of writing HTML, using either `<input>` or `<textarea>`, then having Javascript "replace" them by some rich-text-editor-widget.

But here, we're generating the HTML, so we can of course decide to always use an `<input>` tag and be done with it.

Is it still worth inheriting from `TextInput` then? My gut feeling would be that yes, since that might let us benefit from other things Django provides for `TextInput`. (after all, our widget is an actual `<input type="text">` with some lipstick on it)